File: AtlasModel/Enterprise/TTInterfaceDef.cs
using System;

namespace AtlasModel.Enterprise
{
    public partial class TTInterfaceDef
    {
        public Guid InterfaceDefId { get; set; }
        public string Name { get; set; }
        public string Methods { get; set; }
        public Guid? BaseInterfaceDefId { get; set; }
        public Guid ModuleDefId { get; set; }
        public bool IsBuiltIn { get; set; }
        public string CheckOutId { get; set; }
        public short CheckOutStatus { get; set; }
        public string Description { get; set; }
    }
}

File: AtlasWeb/Core/wwwroot/app/NebulaClient/StorageManager/DefinitionManagement/TTInterfaceDef.ts
import { Guid } from "../../Mscorlib/Guid";
import { TTModuleDef } from "./TTModuleDef";
import { TTReadonlyDictionary } from "./TTReadonlyDictionary";
import { TTInterfaceMemberDef } from "./TTInterfaceMemberDef";
import { TTDefCollection } from "./TTDefCollection";

export class TTInterfaceDef {
    public ModuleDefID: Guid;
    public ModuleDef: TTModuleDef;
    public ID: Guid;
    public InterfaceDefID: Guid;
    public BaseInterfaceDef: TTInterfaceDef;
    public BaseInterfaceDefID: Guid;
    public Name: string;
    public Methods: string;
    public Description: string;
    public IsBuiltIn: boolean;
    public Members: TTReadonlyDictionary<string, TTInterfaceMemberDef>;
    public AllDerivedInterfaceDefs: TTDefCollection<TTInterfaceDef>;

    /*public IsOfType(interfaceDefName: string): boolean {
        return null;
    }
    public IsOfType(interfaceDefID: Guid): boolean {
        return null;
    }
    public IsOfType(interfaceDef: TTInterfaceDef): boolean {
        return null;
    }*/
    public ToString(): string {
        return null;
    }
}
